go/vt/vttablet/tabletmanager/rpc_replication.go

```diff
@@ -747,12 +754,32 @@ func (tm *TabletManager) setReplicationSourceLocked(ctx context.Context, parentA
 	if err != nil {
 		return err
 	}
+
 	host := parent.Tablet.MysqlHostname
 	port := parent.Tablet.MysqlPort
 	// If host is empty, then we shouldn't even attempt the reparent. That tablet has already shutdown.
 	if host == "" {
 		return vterrors.New(vtrpc.Code_FAILED_PRECONDITION, "Shard primary has empty mysql hostname")
 	}
+	// Errant GTID detection.
+	{
+		// Find the executed GTID set of the tablet that we are reparenting to.
+		// We will then compare our own position against it to verify that we don't
+		// have an errant GTID. If we find any GTID that we have, but the primary doesn't,
+		// we will not enter the replication graph and instead fail replication.
+		var primaryPosition string
+		primaryPosition, err = tm.tmc.PrimaryPosition(ctx, parent.Tablet)
```

Comment:
for consistency and simplicity
```suggestion
		primaryPosition, err := tm.tmc.PrimaryPosition(ctx, parent.Tablet)
```

Reply:
Fixed